core/src/main/java/com/sap/ai/sdk/core/AiCoreCredentialsInvalidException.java

```diff
@@ -0,0 +1,7 @@
+package com.sap.ai.sdk.core;
+
+import lombok.experimental.StandardException;
+
+/** Exception thrown when the JSON AI Core service key is invalid. */
+@StandardException
+class AiCoreCredentialsInvalidException extends RuntimeException {}
```

Comment:
([moved from static inner class](https://github.com/SAP/ai-sdk-java/pull/181/files#diff-4da9ee11275233c1ba8615a9f7636b44cc2247b32afb28efa426c143cf545dadL103)) 

The static inner class lead to the name printed out being `com.sap.ai.sdk.core.DestinationResolver$AiCoreCredentialsInvalidException` which isn't great...

Reply:
I would even make it public at this point.

Reply:
Also thought about that, but so far didn't do it because I figured users should never catch this exception explicitly, it's just there to be more descriptive. So no benefit really in making it public, we could also remove it entirely and fully rely on the exception message to do the job. No strong opinion